google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentRunConfigurationEditorTest.java

```diff
@@ -108,6 +110,46 @@ public void testUiAppEngineFlexEnvironment() {
     Disposer.dispose(editor);
   }
 
+  public void testPromote_StopPreviousVersion_Standard() {
+    when(deploymentSource.getEnvironment())
+        .thenReturn(AppEngineEnvironment.APP_ENGINE_STANDARD);
+    AppEngineDeploymentRunConfigurationEditor editor =
+        new AppEngineDeploymentRunConfigurationEditor(
+            getProject(), deploymentSource, appEngineHelper);
+
+    JCheckBox promoteCheckbox = editor.getPromoteCheckbox();
+    JCheckBox stopPreviousVersionCheckbox = editor.getStopPreviousVersionCheckbox();
+
+    assertTrue(promoteCheckbox.isSelected());
+    assertFalse(stopPreviousVersionCheckbox.isVisible());
+
+    Disposer.dispose(editor);
+  }
+
+  public void testPromote_StopPreviousVersion_Flexible() {
+    when(deploymentSource.getEnvironment())
+        .thenReturn(AppEngineEnvironment.APP_ENGINE_FLEX);
+    AppEngineDeploymentRunConfigurationEditor editor =
+        new AppEngineDeploymentRunConfigurationEditor(
+            getProject(), deploymentSource, appEngineHelper);
+
+    JCheckBox promoteCheckbox = editor.getPromoteCheckbox();
+    JCheckBox stopPreviousVersionCheckbox = editor.getStopPreviousVersionCheckbox();
+
+    assertTrue(promoteCheckbox.isSelected());
+    assertTrue(stopPreviousVersionCheckbox.isSelected());
+    assertTrue(stopPreviousVersionCheckbox.isVisible());
+    assertTrue(stopPreviousVersionCheckbox.isEnabled());
+
+    // Disable the promote checkbox and test that stopPreviousVersion behaves correctly
+    promoteCheckbox.setSelected(false);
+
+    assertFalse(stopPreviousVersionCheckbox.isSelected());
+    assertFalse(stopPreviousVersionCheckbox.isEnabled());
+
+    Disposer.dispose(editor);
```

Comment:
editor is defined within the method so that it can be instantiated differently per test